toolchain/base/value_store.h

```diff
@@ -100,6 +101,21 @@ class ValueStore
   auto array_ref() const -> llvm::ArrayRef<ValueType> { return values_; }
   auto size() const -> size_t { return values_.size(); }
 
+  // Makes an iterable range over pairs of the index and value for each value in
+  // the store.
```

Comment:
```suggestion
  // Makes an iterable range over pairs of the index and a reference to the value for each value in
  // the store.
```
I also wonder if it's worth pointing out that
```c++
for (auto [id, value] : store.enumerate())
```
doesn't make copies of `value` -- that's a pretty subtle consequence of the pair holding a reference member, given that `auto` by itself usually does make copies.

Reply:
Good idea, I added a bit more than this to explain.